src/commands/deployWorkspaceProject/getDefaultValues/getDefaultContainerAppsResources/getDefaultContainerAppsResources.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import type { ManagedEnvironment } from "@azure/arm-appcontainers";
+import type { ResourceGroup } from "@azure/arm-resources";
+import { type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { ContainerAppItem, ContainerAppModel } from "../../../../tree/ContainerAppItem";
+import { ManagedEnvironmentItem } from "../../../../tree/ManagedEnvironmentItem";
+import type { DeployWorkspaceProjectSettings } from "../../deployWorkspaceProjectSettings";
+import { getContainerAppResourcesFromItem, getContainerAppResourcesFromSettings } from "./getDefaultResourcesFrom";
+import { promptForEnvironmentResources } from "./promptForEnvironmentResources";
+
+export interface DefaultContainerAppsResources {
+    resourceGroup?: ResourceGroup;
+    managedEnvironment?: ManagedEnvironment;
+    containerApp?: ContainerAppModel;
+}
+
+export async function getDefaultContainerAppsResources(
+    context: ISubscriptionActionContext,
+    settings: DeployWorkspaceProjectSettings | undefined,
+    item?: ContainerAppItem | ManagedEnvironmentItem
+): Promise<DefaultContainerAppsResources> {
+    // Try to obtain container app resources using any saved workspace settings
+    const { resourceGroup, managedEnvironment, containerApp } = await getContainerAppResourcesFromSettings(context, settings);
```

Comment:
Do we want to ignore settings if users right-click a specific item?

I feel like it makes sense to me that the right-clicked item would basically override any workspace setting because I'm specifically choosing this container app.

Maybe we can have a little pop-up that says something like "workspace settings ignored due to selecting container app" or something.

Reply:
Should we maybe add a confirm pop-up instead for these cases?